charts/cluster-scanner/tests/deployment_test.yaml

```diff
@@ -311,3 +313,147 @@ tests:
+          content:
+            name: NATS_TLS_CA
+            valueFrom:
+              configMapKeyRef:
+                name: test-release-cluster-scanner
+                key: js_tls_ca_paths
+                optional: true
+
+  - it: "has volume and volumeMounts if natsjs tls is enabled and custom certs are being provided"
+    set:
+      runtimeStatusIntegrator.natsJS.tls.enabled: true
+      runtimeStatusIntegrator.natsJS.tls.customCerts.existingCaSecret: "testCaSecret"
+      runtimeStatusIntegrator.natsJS.tls.customCerts.existingCaSecretKeyName: "caKeyName"
+      runtimeStatusIntegrator.natsJS.tls.customCerts.existingKeySecret: "testKeySecret"
+      runtimeStatusIntegrator.natsJS.tls.customCerts.existingKeySecretKeyName: "keyKeyName"
+      runtimeStatusIntegrator.natsJS.tls.customCerts.existingCertSecret: "testCertSecret"
+      runtimeStatusIntegrator.natsJS.tls.customCerts.existingCertSecretKeyName: "certKeyName"
+    templates:
+      - ../templates/deployment.yaml
+    asserts:
+      - contains:
+          path: spec.template.spec.volumes
+          content:
+            name: js-certs
+            projected:
+              sources:
+                - secret:
+                    name: testKeySecret
+                    items:
+                      - key: keyKeyName
+                        path: js_tls.key
+                - secret:
+                    name: testCertSecret
+                    items:
+                      - key: certKeyName
+                        path: js_tls.cert
+                - secret:
+                    name: testCaSecret
+                    items:
+                      - key: caKeyName
+                        path: js_tls.ca
+      - contains:
+          path: spec.template.spec.containers[0].volumeMounts
+          content:
+            name: js-certs
+            mountPath: /js-certs
+            readOnly: true
+      - contains:
+          path: spec.template.spec.containers[1].volumeMounts
+          content:
+            name: js-certs
+            mountPath: /js-certs
+            readOnly: true
+      - contains:
+          path: spec.template.spec.containers[1].env
+          content:
+            name: NATS_TLS_CA
+            valueFrom:
+              configMapKeyRef:
+                name: test-release-cluster-scanner
+                key: js_tls_ca_paths
+                optional: true
+
+  - it: "has volume and volumeMounts if natsjs tls is enabled and custom certs are being provided but not the CA Secret"
+    set:
+      runtimeStatusIntegrator.natsJS.tls.enabled: true
+      runtimeStatusIntegrator.natsJS.tls.customCerts.existingKeySecret: "testKeySecret"
+      runtimeStatusIntegrator.natsJS.tls.customCerts.existingKeySecretKeyName: "keyKeyName"
+      runtimeStatusIntegrator.natsJS.tls.customCerts.existingCertSecret: "testCertSecret"
+      runtimeStatusIntegrator.natsJS.tls.customCerts.existingCertSecretKeyName: "certKeyName"
```

Comment:
Can you test when one of the mandatory fields is missing

Reply:
Yep. I added an explicit check for that https://github.com/sysdiglabs/charts/pull/1630/commits/f75da82d0886cc1933706d5a0791976cd6b290a5 . Before these changes the cluster-scanner would misbehave